File: IPtm.py
#!/usr/bin/env python3
import socket
import sys
import argparse
import threading
import time

def ip_transmitter(target_host, target_port, local_port):
    """UDP IP packet forwarder/transmitter"""
    
    # Create UDP socket
    sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
    sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
    
    try:
        sock.bind(('0.0.0.0', local_port))
        print(f"[+] IP Transmitter started: UDP *:{local_port} -> {target_host}:{target_port}")
        print("[+] Waiting for incoming packets...")
        
        while True:
            try:
                data, addr = sock.recvfrom(65535)
                print(f"[+] Fwd {len(data)} bytes from {addr} -> {target_host}:{target_port}")
                
                # Forward exact packet data
                sock.sendto(data, (target_host, target_port))
                
            except KeyboardInterrupt:
                print("\n[!] Interrupted by user")
                break
            except Exception as e:
                print(f"[!] Recv error: {e}")
                time.sleep(0.1)
                
    except Exception as e:
        print(f"[!] Bind failed on port {local_port}: {e}")
        sys.exit(1)
    finally:
        sock.close()

if __name__ == "__main__":
    parser = argparse.ArgumentParser(description="UDP IP Packet Forwarder/Transmitter")
    parser.add_argument("target", help="Target IP:PORT (e.g. 192.168.1.100:12345)")
    parser.add_argument("local_port", type=int, help="Local UDP port to listen on")
    
    args = parser.parse_args()
    
    try:
        target_host, target_port = args.target.split(':')
        target_port = int(target_port)
    except:
        print("Error: target must be IP:PORT format")
        sys.exit(1)
    
    ip_transmitter(target_host, target_port, args.local_port)

File: IPtm.rs
use std::net::{UdpSocket, SocketAddr};
use std::env;
use std::str::FromStr;
use std::thread;
use std::time::Duration;

fn main() {
    let args: Vec<String> = env::args().collect();
    
    if args.len() != 4 {
        eprintln!("Usage: {} <target_ip> <target_port> <local_port>", args[0]);
        eprintln!("Example: {} 192.168.1.100 12345 4444", args[0]);
        std::process::exit(1);
    }

    let target_ip = &args[1];
    let target_port: u16 = args[2].parse().expect("Invalid target port");
    let local_port: u16 = args[3].parse().expect("Invalid local port");

    let target_addr: SocketAddr = format!("{}:{}", target_ip, target_port).parse().expect("Invalid target address");
    
    // Create UDP socket bound to local port
    let socket = UdpSocket::bind(format!("0.0.0.0:{}", local_port))
        .expect("Failed to bind to local port");

    println!("IP Transmitter started on UDP {} -> {}", local_port, target_addr);
    println!("Waiting for incoming packets...");

    loop {
        let mut buf = [0u8; 65535];
        
        match socket.recv_from(&mut buf) {
            Ok((len, src_addr)) => {
                println!("Received {} bytes from {} - forwarding to {}", 
                         len, src_addr, target_addr);
                
                // Forward the exact packet data
                if let Err(e) = socket.send_to(&buf[..len], target_addr) {
                    eprintln!("Failed to forward packet: {}", e);
                }
            }
            Err(e) => {
                eprintln!("Recv error: {}", e);
                thread::sleep(Duration::from_millis(100));
            }
        }
    }
}
